src/test/mocks/MockGsmFailedGetGhoAmountForBuyAsset.sol

```diff
@@ -0,0 +1,582 @@
+    emit TokensRescued(token, to, amount);
+  }
+
+  /// @inheritdoc IGsm
+  function setSwapFreeze(bool enable) external onlyRole(SWAP_FREEZER_ROLE) {
+    if (enable) {
+      require(!_isFrozen, 'GSM_ALREADY_FROZEN');
+    } else {
+      require(_isFrozen, 'GSM_ALREADY_UNFROZEN');
+    }
+    _isFrozen = enable;
+    emit SwapFreeze(msg.sender, enable);
+  }
+
+  /// @inheritdoc IGsm
+  function seize() external notSeized onlyRole(LIQUIDATOR_ROLE) returns (uint256) {
+    _isSeized = true;
+    _currentExposure = 0;
+    _updateExposureCap(0);
+
+    (, uint256 ghoMinted) = IGhoToken(GHO_TOKEN).getFacilitatorBucket(address(this));
+    uint256 underlyingBalance = IERC20(UNDERLYING_ASSET).balanceOf(address(this));
+    if (underlyingBalance > 0) {
+      IERC20(UNDERLYING_ASSET).safeTransfer(_ghoTreasury, underlyingBalance);
+    }
+
+    emit Seized(msg.sender, _ghoTreasury, underlyingBalance, ghoMinted);
+    return underlyingBalance;
+  }
+
+  /// @inheritdoc IGsm
+  function burnAfterSeize(uint256 amount) external onlyRole(LIQUIDATOR_ROLE) returns (uint256) {
+    require(_isSeized, 'GSM_NOT_SEIZED');
+    require(amount > 0, 'INVALID_AMOUNT');
+
+    (, uint256 ghoMinted) = IGhoToken(GHO_TOKEN).getFacilitatorBucket(address(this));
+    if (amount > ghoMinted) {
+      amount = ghoMinted;
+    }
+    IGhoToken(GHO_TOKEN).transferFrom(msg.sender, address(this), amount);
+    IGhoToken(GHO_TOKEN).burn(amount);
+
+    emit BurnAfterSeize(msg.sender, amount, (ghoMinted - amount));
+    return amount;
+  }
+
+  /// @inheritdoc IGsm
+  function updateFeeStrategy(address feeStrategy) external onlyRole(CONFIGURATOR_ROLE) {
+    _updateFeeStrategy(feeStrategy);
+  }
+
+  /// @inheritdoc IGsm
+  function updateExposureCap(uint128 exposureCap) external onlyRole(CONFIGURATOR_ROLE) {
+    _updateExposureCap(exposureCap);
+  }
+
+  /// @inheritdoc IGhoFacilitator
+  function distributeFeesToTreasury() public virtual override {
+    uint256 accruedFees = _accruedFees;
+    if (accruedFees > 0) {
+      _accruedFees = 0;
+      IERC20(GHO_TOKEN).transfer(_ghoTreasury, accruedFees);
+      emit FeesDistributedToTreasury(_ghoTreasury, GHO_TOKEN, accruedFees);
+    }
+  }
+
+  /// @inheritdoc IGhoFacilitator
+  function updateGhoTreasury(address newGhoTreasury) external override onlyRole(CONFIGURATOR_ROLE) {
+    _updateGhoTreasury(newGhoTreasury);
+  }
+
+  /// @inheritdoc IGsm
+  function DOMAIN_SEPARATOR() external view returns (bytes32) {
+    return _domainSeparatorV4();
+  }
+
+  /// @inheritdoc IGsm
+  function getGhoAmountForBuyAsset(
+    uint256 minAssetAmount
+  ) external view returns (uint256, uint256, uint256, uint256) {
+    (
+      uint256 assetAmount,
+      uint256 ghoSold,
+      uint256 grossAmount,
+      uint256 fee
+    ) = _calculateGhoAmountForBuyAsset(minAssetAmount);
+    // TRIGGER ERROR: invalid amount of ghoSold value returned
+    return (assetAmount, ghoSold * 2, grossAmount, fee);
```

Comment:
Instead of having a whole separate mock contract to test this, I think you can just get away with using vm.mockCall() to get this result from the GSM and use it in your tests as desired. Further, if you really only do need something like this, you can have just this function defined in the mock, if you're not calling anything else

Reply:
actually @CheyenneAtapour I remember why I didn't use `mockCall` to begin with, because I needed the mocked call to happen within the converter's invocation of the GSM's `getGhoAmountForSellAsset` rather than from the top level test itself, and as far I know the mockCall won't mock like that. I'll double check again that's the case.

Reply:
nevermind, you were right! Removed it now and just used the mockCall